presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -685,6 +687,111 @@ public static long random(@SqlType(StandardTypes.BIGINT) long value)
         return ThreadLocalRandom.current().nextLong(value);
     }
 
+    @Description("a cryptographically secure random number between 0 and 1 (exclusive)")
+    @ScalarFunction(value = "secure_random", alias = "secure_rand", deterministic = false)
+    @SqlType(StandardTypes.DOUBLE)
+    public static double secure_random()
+    {
+        SecureRandom random = null;
+        try {
+            random = SecureRandom.getInstance("NativePRNGNonBlocking");
+        }
+        catch (NoSuchAlgorithmException e) {
+            random = new SecureRandom();
+        }
+        return random.nextDouble();
+    }
+
+    @Description("a cryptographically secure random number between lower and upper (exclusive)")
+    @ScalarFunction(value = "secure_random", alias = "secure_rand", deterministic = false)
+    @SqlType(StandardTypes.DOUBLE)
+    public static double secure_random(@SqlType(StandardTypes.DOUBLE) double lower, @SqlType(StandardTypes.DOUBLE) double upper)
+    {
+        checkCondition(lower < upper, INVALID_FUNCTION_ARGUMENT, "upper bound must be greater than lower bound");
+        SecureRandom random = null;
+        try {
+            random = SecureRandom.getInstance("NativePRNGNonBlocking");
+        }
+        catch (NoSuchAlgorithmException e) {
+            random = new SecureRandom();
```

Comment:
I'm not sure we want to catch a noSuchAlgorithmException because it means we would basically never know if NativePRNGNonBlocking wasn't being used.  It would be better to throw an internal error.

Reply:
Thanks for reviewing my PR! 
Actually `NativePRNGNonBlocking` is an OS specific secure random implementation. See [this post](https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/). 
I will extract the os-specific secure random algorithm as a static variable then.

For exception handling, I can definitely wrap NoSuchAlgorithmException into a PrestoException and throw it.